### ISSUE TITLE
Deployment: Azure

### ISSUE THREAD

**Issue by tarazena** [vuejs/vue-cli#2186]
### What problem does this feature solve?
Providing instructions on how to deploy Vue-CLI 3.0 app to azure

### What does the proposed API look like?
Since the new Vue CLI 3.0, it introduced a major changes on how the application works, providing a guide for developers to deploy the app to Azure would make sense (already documented on how to do that in AWS and Firebase)

**Comment by tommcclean** [comment]
I was looking for this too, I was able to get Vue projects deployed to Azure before, but don't seem to be able to figure it out with Vue CLI 3

**Comment by tommcclean** [comment]
Has anybody here actually managed to deploy an app using Vue CLI 3>?

Spent all day today and had absolutely no success at all. I can get NPM and everything working with a basic node site - or deploy an older Vue website. But Azure command line cannot recognise vue-cli-service at all.

**Comment by tarazena** [comment]
@tommcclean Yes i was able to deploy Vue CLI 3.0 app to azure, ill write step by step solution on how to do that (I'm using VSTS along with Azure to automate deployments)

**Comment by yauri-io** [comment]
I have succeed building Vue with Vue CLI 3 in Azure.

Remember if you're building in production mode (the NODE_ENV value set to 'production'), it fails with error message vue-cli-service not found.
The problem is because by default, NPM won't install the package in devDependencies (check package.json) when the NODE_ENV set to production mode.

Alternatives to fix the issue:
1. Move necessary packages in devDependencies  to dependencies (I tested but there are many packages I need to move, so I don't use this method)
2. Do not use NODE_ENV=production, instead you can use another name for production. Let's say 'prod' then create Vue environment file with name '.env.prod' and build it 
```
vue-cli-service build --mode prod
```

To read:
- https://medium.com/@stalonadsl948/dependencies-vs-devdependencies-926e096a3dee
- https://cli.vuejs.org/guide/mode-and-env.html

**Comment by datocrats-org** [comment]
Deploy to Azure in this case do you mean to pipeline build on custom vmImage. Microsoft should get their Vue tutorial to work, I'm not sure why it's not working for me but I have tried to reply to this issue asking for their help: https://github.com/MicrosoftDocs/azure-docs/issues/51639 Ideally the vue cli could integrate with Azure like it does with AWS

**Comment by sensedesigns** [comment]
@datocrats-org I can confirm that this solution does not work
